numpy/__init__.py

```diff
@@ -226,16 +320,21 @@ def __getattr__(attr):
         if attr in __former_attrs__:
             raise AttributeError(__former_attrs__[attr])
 
-        if attr == 'testing':
-            import numpy.testing as testing
-            return testing
-
         raise AttributeError("module {!r} has no attribute "
                              "{!r}".format(__name__, attr))
 
     def __dir__():
-        public_symbols = globals().keys() | {'testing'}
-        public_symbols -= {"core", "matrixlib"}
+        # TODO: move away from using `globals` to a statically defined 
+        # list. With `globals`, when running in a testing context 
+        # a bunch of random names fall into global scope, such as
+        # `conftest` or `distutils`.
+        public_symbols = (
+            globals().keys() | __numpy_submodules__
+        )
+        public_symbols -= {
+            "core", "matrixlib", "matlib", "test", "tests", "conftest", 
+            "version", "typing", "f2py", "compat", "array_api", "distutils"
```

Comment:
Some of these are public - `test`, `distutils`, `typing` and `array_api` at least, and those will stay for 2.0.  The others can be left out.

Reply:
Sure! I updated main namespace, so it matches the list from NEP 52.